sim/src/image.rs

```diff
@@ -852,6 +873,50 @@ impl Images {
 
         (flash, resets)
     }
+
+    /// Verify the image in the given flash device, the specified slot
+    /// against the expected image.
+    fn verify_images(&self, flash: &SimMultiFlash, slot: usize, against: usize) -> bool {
+        for image in &self.images {
+            if !verify_image(flash, &image.slots, slot,
+                             match against {
+                                 0 => &image.primaries,
+                                 1 => &image.upgrades,
+                                 _ => panic!("Invalid 'against'"),
+                             }) {
+                return false;
+            }
+        }
+        true
+    }
+
+    /// Verify that the trailers of the images have the specified
+    /// values.
+    fn verify_trailers(&self, flash: &SimMultiFlash, slot: usize,
```

Comment:
It might be possible that checking both images together in those tests might break some of the testing that expects errors, etc. This applies here and for `verify_images` above, so it's good to be aware. For mono images this should remain working as before, so looks good!

Reply:
These loops essentially are an 'and' condition. It at least seems to work for all of the current tests with multiple images. It will get an error if either of the images is incorrect.